test/framework/src/main/java/org/elasticsearch/index/mapper/FieldMapperTestCase.java

```diff
@@ -205,4 +203,47 @@ protected String contentType() {
         }
     }
 
+    public void testSerialization() throws IOException {
+        for (Modifier modifier : modifiers) {
+            if (unsupportedProperties().contains(modifier.property)) {
+                continue;
+            }
+            T builder1 = newBuilder();
+            T builder2 = newBuilder();
+            modifier.apply(builder1, builder2);
+            assertSerializes(modifier.property + "-a", builder1);
+            assertSerializes(modifier.property + "-b", builder2);
+        }
+    }
+
+    protected Settings getIndexMapperSettings() {
+        return Settings.EMPTY;
+    }
+
+    protected void assertSerializes(String indexname, T builder) throws IOException {
+
+        // TODO can we do this without building an entire index?
```

Comment:
This is a bit slow, unfortunately - the idea is to check that serializing a mapper and then using it as a merge input always ends up with the same mapper, for each possible modification.  But we need a new MapperService for each test, because some modifications are unmergeable.  And that means we have to rebuild a whole new index each time, so that we load mappers from plugins, etc.  Which is slow.